osconfig_tests/osconfig_server/osconfig_service_wrapper.go

```diff
@@ -0,0 +1,93 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package contains wrapper around osconfig service APIs and helper methods
+package osconfig_server
+
+import (
+	"context"
+	"log"
+
+	osconfig "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/cloud.google.com/go/osconfig/apiv1alpha1"
+	osconfigpb "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha1"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/google-osconfig-agent/config"
+	"github.com/kylelemons/godebug/pretty"
+	"google.golang.org/api/option"
+)
+
+var dump = &pretty.Config{IncludeUnexported: true}
+
+func getOsConfigClient(ctx context.Context, logger *log.Logger) (*osconfig.Client, error) {
+	client, err := osconfig.NewClient(ctx, option.WithEndpoint(config.SvcEndpoint()), option.WithCredentialsFile(config.OAuthPath()))
+
+	if err != nil {
+		logger.Printf("error while creating osconfig client: %s\n", err)
+	}
+	return client, err
+
+}
+
+func CreateOsConfig(ctx context.Context, logger *log.Logger, req *osconfigpb.CreateOsConfigRequest) (*osconfigpb.OsConfig, error) {
+	client, err := getOsConfigClient(ctx, logger)
+
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Printf("create osconfig request:\n%s\n\n", dump.Sprint(req))
+
+	res, err := client.CreateOsConfig(ctx, req)
+	if err != nil {
+		logger.Printf("error while creating osconfig:\n%s\n\n", err)
+		return nil, err
+	}
+	logger.Printf("create osconfig response:\n%s\n\n", dump.Sprint(res))
+
+	return res, nil
+}
+
+func ListOsConfigs(ctx context.Context, logger *log.Logger, req *osconfigpb.ListOsConfigsRequest) *osconfig.OsConfigIterator {
+	client, err := getOsConfigClient(ctx, logger)
```

Comment:
Is there a reason to recreate the client each time instead of using a singleton client? I'm not advocating any change, just curious.